tests/harness/cases/optional.proto

```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+package tests.harness.cases;
+option go_package = "github.com/envoyproxy/protoc-gen-validate/tests/harness/cases/go;cases";
+
+import "validate/validate.proto";
+
+message TestOptionalMsg {
+    string  const  = 1 [(validate.rules).string.const = "foo"];
+    TestOptionalMsg nested = 2;
+}
+
+message MessageRequiredButOptional { optional TestOptionalMsg val = 1 [(validate.rules).message.required = true]; }
```

Comment:
the build in failing for java because it says that `the value is optional by default` so you don't need the keyword `optional`. Upon removing it from my local build, the build passes.